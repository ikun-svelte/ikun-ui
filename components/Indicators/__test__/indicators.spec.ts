import { afterEach, beforeEach, describe, expect, test, vi } from 'vitest';
import KIndicators from '../src';
import KIndicatorsSlot from './fixture/slots.svelte';
import { tick } from 'svelte';
import { fireEvent } from '@testing-library/svelte';
let host;

const initHost = () => {
	host = globalThis.document.createElement('div');
	host.setAttribute('id', 'host');
	globalThis.document.body.appendChild(host);
};
beforeEach(() => {
	initHost();
	vi.useFakeTimers();
});
afterEach(() => {
	host.remove();
	vi.useRealTimers();
});

describe('Test: KIndicators', () => {
	test('props: cls', async () => {
		const instance = new KIndicators({
			target: host,
			props: {
				cls: 'k-indicators--test'
			}
		});
		expect(instance).toBeTruthy();
		expect(host!.innerHTML.includes('k-indicators--test')).toBeTruthy();
		expect(host.innerHTML).matchSnapshot();
	});

	test('props: defaultPageIndex', async () => {
		const instance = new KIndicators({
			target: host,
			props: {
				defaultPageIndex: 4,
				count: 6
			}
		});
		expect(instance).toBeTruthy();
		expect(host.innerHTML).matchSnapshot();
	});

	test('props: count', async () => {
		const instance = new KIndicators({
			target: host,
			props: {
				count: 6
			}
		});
		expect(instance).toBeTruthy();
		expect(host.innerHTML).matchSnapshot();
	});

	test('props: trigger is click', async () => {
		let index = 0;
		const instance = new KIndicators({
			target: host,
			props: {
				count: 6,
				trigger: 'click'
			}
		});
		expect(instance).toBeTruthy();
		instance.$on('change', (e: CustomEvent) => {
			index = e.detail.index;
		});
		const btn = host.querySelectorAll('.k-indicators-item')[3];
		btn.click();
		await tick();
		expect(index).toBe(3);
	});

	test('props: trigger is hover', async () => {
		let index = 0;
		const instance = new KIndicators({
			target: host,
			props: {
				count: 6,
				trigger: 'hover'
			}
		});
		expect(instance).toBeTruthy();
		instance.$on('change', (e: CustomEvent) => {
			index = e.detail.index;
		});
		const btn = host.querySelectorAll('.k-indicators-item')[3];
		await fireEvent.mouseEnter(btn);
		await tick();
		expect(index).toBe(3);
	});

	test('slots: default', async () => {
		const instance = new KIndicatorsSlot({
			target: host
		});
		expect(instance).toBeTruthy();
		expect(host.innerHTML).matchSnapshot();
	});
});
