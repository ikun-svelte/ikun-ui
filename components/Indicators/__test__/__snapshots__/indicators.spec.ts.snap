// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Test: KIndicators > props: cls 1`] = `"<div class="k-indicators k-indicators--test"></div>"`;

exports[`Test: KIndicators > props: count 1`] = `"<div class="k-indicators"><div role="button" class="k-indicators-item k-indicators-item--active" aria-hidden="true"></div><div role="button" class="k-indicators-item" aria-hidden="true"></div><div role="button" class="k-indicators-item" aria-hidden="true"></div><div role="button" class="k-indicators-item" aria-hidden="true"></div><div role="button" class="k-indicators-item" aria-hidden="true"></div><div role="button" class="k-indicators-item" aria-hidden="true"></div></div>"`;

exports[`Test: KIndicators > props: defaultPageIndex 1`] = `"<div class="k-indicators"><div role="button" class="k-indicators-item" aria-hidden="true"></div><div role="button" class="k-indicators-item" aria-hidden="true"></div><div role="button" class="k-indicators-item" aria-hidden="true"></div><div role="button" class="k-indicators-item" aria-hidden="true"></div><div role="button" class="k-indicators-item k-indicators-item--active" aria-hidden="true"></div><div role="button" class="k-indicators-item" aria-hidden="true"></div></div>"`;

exports[`Test: KIndicators > slots: default 1`] = `"<div class="k-indicators"><div slot="item">0</div></div>"`;
